core/src/main/java/com/google/zxing/qrcode/detector/FinderPatternFinder.java

```diff
@@ -604,85 +614,60 @@ private boolean haveMultiplyConfirmedCenters() {
       throw NotFoundException.getNotFoundInstance();
     }
 
-    // Filter outlier possibilities whose module size is too different
-    if (startSize > 3) {
-      // But we can only afford to do so if we have at least 4 possibilities to choose from
-      double totalModuleSize = 0.0;
-      double square = 0.0;
-      for (FinderPattern center : possibleCenters) {
-        float size = center.getEstimatedModuleSize();
-        totalModuleSize += size;
-        square += (double) size * size;
-      }
-      double average = totalModuleSize / startSize;
-      float stdDev = (float) Math.sqrt(square / startSize - average * average);
+    Collections.sort(possibleCenters, moduleComparator);
 
-      Collections.sort(possibleCenters, new FurthestFromAverageComparator((float) average));
+    double distortion = Double.MAX_VALUE;
+    double[] squares = new double[3];
+    FinderPattern[] bestPatterns = new FinderPattern[3];
 
-      float limit = Math.max(0.2f * (float) average, stdDev);
+    for (int i = 0; i < possibleCenters.size() - 2; i++) {
+      float minModuleSize = possibleCenters.get(i).getEstimatedModuleSize();
```

Comment:
It may not matter much, but was also thinking you can save the reference to `possibleCenters.get(...)` outside the loop to avoid repeatedly accessing the field and indexing into the collection.

Reply:
Do you mean to call toArray before the loop?

Reply:
No, I mean things like:
```
FinderPatternFinder fpi = possibleCenters.get(i);
```
outside the j/k loops for example.
It's not a big deal but this part is performance sensitive.

Reply:
I see. I fixed it.